File: robot/Vision/SegmentedImage/RGB.cpp
#include "RGB.h"

RGB::RGB(unsigned char _red, unsigned char _green, unsigned char _blue)
  : red(_red),
    green(_green),
    blue(_blue) {
}

RGB::~RGB() {
}

void RGB::set(unsigned char _red, unsigned char _green, unsigned char _blue) {
  red   = _red;
  green = _green;
  blue  = _blue;
}

int RGB::getColor() const {
  return (((int)red   << 16) & 0xFF0000)
       | (((int)green << 8 ) & 0x00FF00)
       | (((int)blue       ) & 0x0000FF);
}

unsigned char RGB::getRed() const {
  return red;
}

unsigned char RGB::getGreen() const {
  return green;
}

unsigned char RGB::getBlue() const {
  return blue;
}

File: tools/SharedJavaLibrary/src/image/format/RGB.java
package image.format;

public class RGB {

  public RGB(int red, int green, int blue) {
    set(red, green, blue);
  }
  
  public int getRed() {
    return red;
  }
  
  public int getGreen() {
    return green;
  }
  
  public int getBlue() {
    return blue;
  }
  
  public int toInt() {
    return ((red & 0xFF) << 16) |
           ((green & 0xFF) << 8) |
            (blue & 0xFF);
  }
  
  public void set(int red, int green, int blue) {
    if ((red < 0) || (red > 255)) {
      throw new IllegalArgumentException("Red should be within 0 and 255");
    }
    if ((green < 0) || (green > 255)) {
      throw new IllegalArgumentException("Green should be within 0 and 255");
    }
    if ((blue < 0) || (blue > 255)) {
      throw new IllegalArgumentException("Blue should be within 0 and 255");
    }

    this.red   = red;
    this.green = green;
    this.blue  = blue;
  }
  
  private int red, green, blue;
  
}
